### ISSUE TITLE
Cannot use combo as cell editor template

### ISSUE THREAD

**Issue by wnvko** [IgniteUI/igniteui-react#62]
## Description  
I want to use combo as cell editor template in the grid.

## Steps to reproduce  
Add a combo as cell editor template in the grid. Try to update the cell value. I am using code like this:
```tsx
const comboInlineEditorTemplate = (ctx: { dataContext: IgrCellTemplateContext }) => {
  return (
    <>
      <IgrCombo data={friends}
        value={ctx.dataContext.cell.value}
        change={(_s: IgrCombo, e: IgrComboChangeEventArgs) => ctx.dataContext.cell.editValue = e.detail.newValue}
        displayKey="first_name"
        valueKey="id">
      </IgrCombo>
    </>
  )
}

return (
  <>
    <div className={classes("column-layout master-view-container")}>
      <IgrGrid data={people} primaryKey="id" rowEditable="true" autoGenerate="false">
        <IgrColumn field="friends" inlineEditorTemplate={comboInlineEditorTemplate}></IgrColumn>
      </IgrGrid>
    </div>
  </>
);

```

## Result  
When I put the cell in edit mode the combo appears. I can drop down the combo but not able to select/deselect any values in the combo with the mouse.
If I try to navigate with the keyboard I am able to reach any item and select/deselect it with space key.
Even if I change the selection of the combo, with the keyboard, when I press `enter` old selection returns as value of the combo and I am not able to update the cell value.

## Expected result  
Should be able to use `IgrCombo` as editor in cell edit template and should be able to update the cell value.

**Comment by dkamburov** [comment]
Cherry-picked to blazor branch - https://github.com/IgniteUI/igniteui-angular/pull/12125/commits/758c8fc4962725964da53f021320812b1ebfabeb

**Comment by MayaKirova** [comment]
@wnvko Fix is available in internal build 23.2.607 and will be available in the next public release.
